### PR TITLE
[Navigation] Import new navigation from auth

### PR DESCRIPTION
**Pull request by blazarecki**

### CODE REVIEW

File: src/Controller/RedirectingController.php [hunk 1]
```diff
@@ -21,4 +21,21 @@ public function removeTrailingSlashAction(Request $request)
 
         return $this->redirect($url, Response::HTTP_MOVED_PERMANENTLY);
     }
+
+    public function urlRedirectAction(Request $request, $path): Response
```
Comment (B-Galati) [inline src/Controller/RedirectingController.php]: Looks unused no ?

File: templates/adherent/_profile_menu.html.twig [hunk 2]
```diff
@@ -2,15 +2,23 @@
...
+            'app_user_set_email_notifications': 'Notifications',
         } %}
             <li class="{{ route == target ? 'active' }}">
                 <a href="{{ path(target) }}">{{ link }}</a>
             </li>
         {% endfor %}
+
+        {% if is_granted('ROLE_ADHERENT') %}
+            {% for target, link in {
+                'app_summary_manager_index': 'Modifier mon profil',
```
Comment (B-Galati) [inline templates/adherent/_profile_menu.html.twig]: It looks like we don't need the loop anymore no ?

File: templates/adherent/home.html.twig [hunk 3]
```diff
@@ -0,0 +1,31 @@
+{% extends 'base.html.twig' %}
+
+{% block page_title 'Bienvenue, ' ~ app.user.firstName %}
+
+{% block content %}
+    <main>
+        <section class="content b__nudge--bottom-huge">
+            <header class="l__wrapper--slim space--60-0">
+                <h1 class="text--center text--large">
+                    Bienvenue, {{ app.user.firstName }} ! 👋
```
Comment (B-Galati) [inline templates/adherent/home.html.twig]: I don't know anything about this. Is it ok to use unicode emoji directly in the HTML ? Are they going to be rendered correctly in most cases ?

File: templates/citizen_project/_sidebar.html.twig [hunk 4]
```diff
@@ -10,8 +10,8 @@
         <h3>Nous recherchons</h3>
         <ul class="citizen-project__skills">
             {% if citizen_project.skills|length > 0 %}
-                {% for skill in citizen_project.skills %}
-                    <li>{{ skill }}</li>
+                {% for skill in citizen_project.skills  %}
```
Comment (B-Galati) [inline templates/citizen_project/_sidebar.html.twig]: It can cause useless conflicts

File: templates/adherent/home.html.twig (hunk 3 above)
Comment (marineb) [reply templates/adherent/home.html.twig]: Yeah. It is, plus it's old code.

File: front/style/layout/_nav.scss [hunk 5]
```diff
@@ -1,198 +1,210 @@
...
-                    }
-                }
+            &:last-child {
+                padding: 3px 0 0 0;
             }
 
-            &__account {
-                padding: 20px 25px;
-                font-family: $font-roboto;
+            // &.bullet:before {
```
Comment (HeahDude) [inline front/style/layout/_nav.scss]: We should not commit commented code

File: src/Controller/EnMarche/UserController.php [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+use Symfony\Component\Form\Extension\Core\Type\SubmitType;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * @Route("/parametres/mon-compte")
+ */
+class UserController extends Controller
+{
+    const UNREGISTER_TOKEN = 'unregister_token';
```
Comment (Remg) [inline src/Controller/EnMarche/UserController.php]: missing visibility

File: templates/adherent/_membership_form.html.twig [hunk 7]
```diff
@@ -64,11 +64,11 @@
...
         <div class="l__col l__col--top register__form--third" id="field-birthdate">
             {{ form_label(form.birthdate, 'Date de naissance', { 'label_attr': {'class': 'text--gray'} }) }}
-           <div class="l__row l__row--h-stretch">
-               {% for field in form.birthdate|reverse %}
-                   {{ form_widget(field, {'attr': {'style': 'width: 32%;'}}) }}
-               {% endfor %}
-           </div>
+            <div class="l__row l__row--h-stretch">
+                {% for field in form.birthdate|reverse %}
+                    {{ form_widget(field, {'attr': {'style': 'width: 32%;'}}) }}
```
Comment (Remg) [inline templates/adherent/_membership_form.html.twig]: `{ attr: { style: 'width: 32%;' } }`

File: templates/adherent/home.html.twig (hunk 3 above)
Comment (HeahDude) [reply templates/adherent/home.html.twig]: Yes it's unicode.

File: templates/adherent/home.html.twig [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+                </h1>
+            </header>
+
+            <article class="l__wrapper--slim">
+                <p class="text--body">
+                    Merci encore de nous avoir rejoint ! Voici quelques action à votre disposition dés maintenant.
```
Comment (HeahDude) [inline templates/adherent/home.html.twig]: Typo: actions, dès

Comment (Remg) [inline templates/adherent/home.html.twig]: quelques action**s**

File: templates/base.html.twig [hunk 9]
```diff
@@ -101,443 +102,30 @@
 
         {% block header %}
         <header>
+            {% set adherent_committees = '' %}
+            {# Prerendering this block and saving it in a variable prevents extra SQL queries #}
+            {% if is_granted('ROLE_ADHERENT') %}
```
Comment (HeahDude) [inline templates/base.html.twig]: `has_role_adherent`

File: templates/adherent/list_my_committees_al.html.twig [hunk 10]
```diff
@@ -0,0 +1,13 @@
+<ul class="list__links">
+    {% for committee in committees %}
+        {% if app.user.hostOf(committee) %}
+            {# using the Adherent::isHostOf() method instead of a voter below prevents unneeded SQL queries #}
+            <li>
+                <a href="{{ path('app_committee_show', {'slug': committee.slug}) }}"
```
Comment (Remg) [inline templates/adherent/list_my_committees_al.html.twig]: `{ slug: committee.slug }`

File: templates/adherent/profile.html.twig [hunk 11]
```diff
@@ -3,10 +3,11 @@
 {% use 'adherent/_membership_form.html.twig' %}
 
 {% block javascripts %}
+    <script src="https://maps.googleapis.com/maps/api/js?key={{ google_maps_api_key }}&libraries=places"></script>
     {% import 'javascript.js.twig' as js %}
     <script type="text/javascript">
         Kernel.onLoad(function() {
-            {{ js.address(form.address) }}
+            {{ js.address(form.address, {cityAutocomplete: true}) }}
```
Comment (Remg) [inline templates/adherent/profile.html.twig]: `{ cityAutocomplete: true }`

File: templates/components/_footer.html.twig [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+        </div>
+        <div class="l__row l__row--h-stretch l__row--top b__nudge--bottom-large l__row--wrap l__mobile--col">
+            <ul class="list__links list__links--col list__links--no-decor">
+                <li>
+                    <h4>
+                        <a href="{{ path('page_le_mouvement') }}" class="text--bold text--medium-small">Le Mouvement</a>
+                    </h4>
+                </li>
+                <li>
+                    <a href="{{ path('page_le_mouvement') }}">Qui sommes-nous ?</a>
```
Comment (HeahDude) [inline templates/components/_footer.html.twig]: Same as above, intended?

File: templates/components/_footer.html.twig [hunk 13]
```diff
@@ -0,0 +1,124 @@
...
+                </li>
+            </ul>
+            <ul class="list__links list__links--col list__links--no-decor">
+                <li>
+                    <h4>
+                        <a href="{{ path('page_campus') }}" class="text--bold text--medium-small">Campus</a>
+                    </h4>
+                </li>
+                <li>
+                    <a href="{{ path('page_campus') }}">Toutes les ressources</a>
```
Comment (HeahDude) [inline templates/components/_footer.html.twig]: Same link as above, intended?

File: templates/components/_footer.html.twig [hunk 14]
```diff
@@ -0,0 +1,124 @@
...
+                    <li><h4><a href="#" class="text--bold text--medium-small">S'engager</a></h4></li>
+                {% endif %}
+                    <li><a href="{{ path('donation_index') }}">Donner en ligne</a></li>
+                    <li><a href="https://storage.googleapis.com/en-marche-fr/bulletin-de-don-EMa.pdf">Donner par chèque</a></li>
+                {% if not campaign_expired %}
+                    <li><a href="{{ path('app_search_committees') }}">Rejoindre un comité</a></li>
+                    <li><a href="{{ path('app_search_events') }}">S'inscrire à un événement</a></li>
+                    <li><a href="{{ path('invitation_form') }}">Inviter une connaissance</a></li>
+                    <li>
+                        <a href="{{ is_granted('ROLE_ADHERENT') ? path('app_user_profile') : path('app_user_login') }}">
```
Comment (HeahDude) [inline templates/components/_footer.html.twig]: `has_role_adherent`

File: templates/components/_nav_desktop.html.twig [hunk 15]
```diff
@@ -0,0 +1,214 @@
+<nav class="gtm__navigation__desktop l__wrapper l__row l__row--h-stretch l__hide--on-small-laptop {{ app.user ? '' : 'space--20-0' }}">
+    <ul class="list__links list__links--row list__links--no-decor list__links--nav">
+        {% if app.user %}
```
Comment (HeahDude) [inline templates/components/_nav_desktop.html.twig]: `has_role_adherent`

File: templates/components/_nav_desktop.html.twig [hunk 16]
```diff
@@ -0,0 +1,214 @@
...
+        <li>
+            <h1><a href="{{ path('homepage') }}">EM!</a></h1>
+        </li>
+        {% endif %}
+        {% if not app.user %}
```
Comment (HeahDude) [inline templates/components/_nav_desktop.html.twig]: `else`

File: templates/components/_nav_desktop.html.twig [hunk 17]
```diff
@@ -0,0 +1,214 @@
...
+        <li>
+            <a href="{{ path('app_citizen_project_landing') }}">Projets citoyens</a>
+        </li>
+        <li>
+            <a href="{{ path('page_campus') }}">Campus</a>
+        </li>
+        <li>
+            <a href="{{ path('articles_list', {'category': 'tout'}) }}">Actualités</a>
+        </li>
+        {% if app.user and not is_granted('ROLE_ADHERENT') %}
```
Comment (HeahDude) [inline templates/components/_nav_desktop.html.twig]: `not has_role_adherent`

File: templates/components/_nav_desktop.html.twig [hunk 18]
```diff
@@ -0,0 +1,214 @@
...
+                        Admin 🤓
+                    </a>
+                {% else %}
+                    {% if app.user %}
+                        <a href="{{ is_granted('ROLE_ADHERENT') ? path('app_user_profile') }}" class="icon--with-text nav-dropdown__name">
+                            {{ include('components/user.html.twig') }}
+                            {{ app.user.firstName is defined ? app.user.firstName : ''}}
+                        </a>
+                    {% endif %}
+                    {% if is_granted('ROLE_USER') %}
```
Comment (HeahDude) [inline templates/components/_nav_desktop.html.twig]: `has_role_user`

File: templates/components/_nav_desktop.html.twig [hunk 19]
```diff
@@ -0,0 +1,214 @@
...
+                            <span class="nav-dropdown__caret--outer"></span>
+                            <span class="nav-dropdown__caret--inner"></span>
+                        </div>
+                        <div class="nav-dropdown__menu">
+                            <ul class="nav-dropdown__section">
+                                <li class="nav-dropdown__section__name">
+                                    <a href="{{ is_granted('ROLE_ADHERENT') ? path('app_user_profile') }}" class="icon--with-text">
+                                        <div>{{ app.user }}</div>
+                                        <div class="text--small text--gray text--normal">
+                                            {% if not is_granted('ROLE_ADHERENT') %}
```
Comment (HeahDude) [inline templates/components/_nav_desktop.html.twig]: `has_role_adherent` same below, and would need a `has_role_referent` too.

File: templates/components/_nav_desktop.html.twig [hunk 20]
```diff
@@ -0,0 +1,214 @@
...
+                                        </div>
+                                    </a>
+                                </li>
+                            </ul>
+                            <div class="nav-dropdown__separator"></div>
+                            <div class="nav-dropdown__row">
+                                <div>
+                                    <ul class="nav-dropdown__section">
+                                        <li>
+                                            {% set url = is_granted('ROLE_ADHERENT') ? path('app_user_profile') : path('app_user_login') %}
```
Comment (HeahDude) [inline templates/components/_nav_desktop.html.twig]: Same here and below.

File: templates/components/_nav_desktop.html.twig [hunk 21]
```diff
@@ -0,0 +1,214 @@
...
+                    {% endif %}
+                {% endif %}
+            </div>
+            {% if (not app.user) %}
+                <a href="{{ path('app_user_login') }}" class="text--body link--no-decor b__nudge--right">Connexion</a>
+                <a href="{{ path('app_membership_register') }}" class="btn btn--blue">Inscription</a>
+            {% endif %}
+    </div>
+</nav>
+{% if is_granted('ROLE_SUPERVISOR') or is_granted('ROLE_HOST') or is_granted('ROLE_REFERENT') or is_granted('ROLE_BOARD_MEMBER') or is_granted('ROLE_COORDINATOR') or is_granted('ROLE_PROCURATION_MANAGER') or is_granted('ROLE_LEGISLATIVE_CANDIDATE') %}
```
Comment (HeahDude) [inline templates/components/_nav_desktop.html.twig]: There far too many usage of `is_granted` around here.

File: templates/components/_nav_mobile.html.twig [hunk 22]
```diff
@@ -0,0 +1,212 @@
...
+                    <li>
+                        <a href="{{ path('app_membership_register') }}">Inscription</a>
+                    </li>
+                </ul>
+                <hr />
+            {% endif %}
+            {% if is_granted('ROLE_USER') %}
+                <ul>
+                    <li>
+                        <a href="{{ is_granted('ROLE_ADHERENT') ? path('app_user_profile') : path('app_user_login') }}">
```
Comment (HeahDude) [inline templates/components/_nav_mobile.html.twig]: Same in this file.

File: templates/components/_footer.html.twig [hunk 23]
```diff
@@ -0,0 +1,124 @@
...
+        </div>
+        <div class="l__row l__row--h-stretch l__tablet--col">
+            <div class="donate-mail">
+                Vous pouvez envoyer vos dons en
+                <a href="https://storage.googleapis.com/en-marche-fr/bulletin-de-don-EMa.pdf" class="text--bold" target="_blank" rel="noopener noreferrer">téléchargeant ce bulletin</a>.
+            </div>
+            <ul class="list__links list__links--row list__links--no-decor list__links--col-mobile">
+                <li><a href="https://aide.en-marche.fr" target="_blank" rel="noopener noreferrer">Aide</a></li>
+                <li><a href="https://contact.en-marche.fr" target="_blank" rel="noopener noreferrer">Écrivez-nous</a></li>
+                <li><a href="{{ path('articles_list', {category: 'communiques'}) }}">Presse</a></li>
```
Comment (Remg) [inline templates/components/_footer.html.twig]: `{ category: 'communiques' }`

File: templates/user/my_account.html.twig [hunk 24]
```diff
@@ -0,0 +1,33 @@
...
+    </div>
+
+    {# This will be enable by the PR on branch auth2 #}
+    {#{% if is_granted('ROLE_ADHERENT') and not is_granted('ROLE_REFERENT') and not is_granted('ROLE_ADMINISTRATOR') %}#}
+        {#<p>#}
+            {#<a href="{{ path('app_user_terminate_membership') }}" class="settings__delete_account">Annuler mon adhésion</a>#}
+        {#</p>#}
+    {#{% endif %}#}
+
+    {% if is_granted('UNREGISTER') and app.user.isBasicAdherent() %}
```
Comment (HeahDude) [inline templates/user/my_account.html.twig]: `app.user.basicAdherent`

File: templates/components/_nav_desktop.html.twig (hunk 16 above)
Comment (Remg) [inline templates/components/_nav_desktop.html.twig]:
```
-        {% endif %}
-        {% if not app.user %}
+        {% else %}
```

File: templates/components/_nav_desktop.html.twig [hunk 25]
```diff
@@ -0,0 +1,214 @@
...
+            <a href="{{ path('app_search_events') }}">Événements</a>
+        </li>
+        <li>
+            <a href="{{ path('app_citizen_project_landing') }}">Projets citoyens</a>
+        </li>
+        <li>
+            <a href="{{ path('page_campus') }}">Campus</a>
+        </li>
+        <li>
+            <a href="{{ path('articles_list', {'category': 'tout'}) }}">Actualités</a>
```
Comment (Remg) [inline templates/components/_nav_desktop.html.twig]: `{ category: 'tout' }`

File: templates/components/_footer.html.twig (hunk 12 above)
Comment (B-Galati) [reply templates/components/_footer.html.twig]: I think so, @marineb can you confirm that point ?

File: templates/components/_footer.html.twig (hunk 13 above)
Comment (B-Galati) [reply templates/components/_footer.html.twig]: same, can you confirm it is intended @marineb ?

File: templates/components/_footer.html.twig (hunk 12 above)
Comment (Remg) [reply templates/components/_footer.html.twig]: @marineb

File: templates/user/my_account.html.twig (hunk 24 above)
Comment (B-Galati) [reply templates/user/my_account.html.twig]: I prefer the version above but ok

File: templates/components/_nav_desktop.html.twig (hunk 15 above)
Comment (Remg) [reply templates/components/_nav_desktop.html.twig]: nope, a user doesnt have to adhere anymore \o/

File: templates/adherent/home.html.twig (hunk 3 above)
Comment (B-Galati) [reply templates/adherent/home.html.twig]: thanks ;-)

**Comment by B-Galati** [comment]
@HeahDude @Remg @blazarecki most of the comments are fixed. thanks !

File: templates/adherent/home.html.twig [hunk 26]
```diff
@@ -0,0 +1,30 @@
...
+        <section class="content b__nudge--bottom-huge">
+            <header class="l__wrapper--slim space--60-0">
+                <h1 class="text--center text--large">
+                    Bienvenue, {{ app.user.firstName }} ! 👋
+                </h1>
+            </header>
+
+            <article class="l__wrapper--slim">
+                <p class="text--body">
+                    Merci encore de nous avoir rejoint ! Voici quelques actionc à votre disposition dès maintenant.
```
Comment (Remg) [inline templates/adherent/home.html.twig]: typo

File: templates/components/_nav_mobile.html.twig [hunk 27]
```diff
@@ -0,0 +1,212 @@
...
+                    <a href="{{ path('app_search_events') }}">Événements</a>
+                </li>
+                <li>
+                    <a href="{{ path('app_citizen_project_landing') }}">Projets citoyens</a>
+                </li>
+                <li>
+                    <a href="{{ path('page_campus') }}">Campus</a>
+                </li>
+                <li>
+                    <a href="{{ path('articles_list', {'category': 'tout'}) }}">Actualités</a>
```
Comment (thomasbisignani) [inline templates/components/_nav_mobile.html.twig]: You can remove simple quotes for keys here : `{ category: 'tout' }`

File: templates/search/_citizen_projects.html.twig [hunk 28]
```diff
@@ -21,15 +21,15 @@
                 {% if citizen_project.skills|length > 0 %}
                 Nous recherchons
                     <ul class="search__citizen_project__skills">
-                        {% for skill in citizen_project.skills %}
+                        {% for skill in citizen_project.skills  %}
```
Comment (thomasbisignani) [inline templates/search/_citizen_projects.html.twig]: white space useless here

File: templates/components/_nav_desktop.html.twig (hunk 15 above)
Comment (HeahDude) [reply templates/components/_nav_desktop.html.twig]: `has_role_user` then :p

File: templates/components/_nav_desktop.html.twig [hunk 29]
```diff
@@ -0,0 +1,213 @@
...
+                        Admin 🤓
+                    </a>
+                {% else %}
+                    {% if app.user %}
+                        <a href="{{ has_role_adherent ? path('app_user_profile') }}" class="icon--with-text nav-dropdown__name">
+                            {{ include('components/user.html.twig') }}
+                            {{ app.user.firstName is defined ? app.user.firstName : ''}}
+                        </a>
+                    {% endif %}
+                    {% if has_role_user %}
```
Comment (HeahDude) [inline templates/components/_nav_desktop.html.twig]: This if can be merged with above, cannot it?

File: templates/components/_nav_desktop.html.twig [hunk 30]
```diff
@@ -0,0 +1,213 @@
...
+                        </div>
+                        <div class="nav-dropdown__menu">
+                            <ul class="nav-dropdown__section">
+                                <li class="nav-dropdown__section__name">
+                                    <a href="{{ has_role_adherent ? path('app_user_profile') }}" class="icon--with-text">
+                                        <div>{{ app.user }}</div>
+                                        <div class="text--small text--gray text--normal">
+                                            {% if not has_role_adherent %}
+                                                Non-adhérent(e)
+                                            {% elseif is_granted('ROLE_REFERENT') %}
```
Comment (HeahDude) [inline templates/components/_nav_desktop.html.twig]: `app.user.isReferent`

File: templates/components/_nav_desktop.html.twig [hunk 31]
```diff
@@ -0,0 +1,213 @@
...
+                                <div>
+                                    <ul class="nav-dropdown__section">
+                                        <li>
+                                            {% set url = has_role_adherent ? path('app_user_profile') : path('app_user_login') %}
+                                            <a href="{{ url }}">Profil</a>
+                                        </li>
+                                        <li>
+                                            <a href="{{ path('app_adherent_events') }}">Mes activités</a>
+                                        </li>
+                                        {% if app.user and (has_role_adherent or is_granted('ROLE_HOST')) %}
```
Comment (HeahDude) [inline templates/components/_nav_desktop.html.twig]: `app.user.isHost`

File: templates/components/_nav_desktop.html.twig [hunk 32]
```diff
@@ -0,0 +1,213 @@
...
+            {% if (not app.user) %}
+                <a href="{{ path('app_user_login') }}" class="text--body link--no-decor b__nudge--right">Connexion</a>
+                <a href="{{ path('app_membership_register') }}" class="btn btn--blue">Inscription</a>
+            {% endif %}
+    </div>
+</nav>
+{% if is_granted('ROLE_SUPERVISOR') or is_granted('ROLE_HOST') or is_granted('ROLE_REFERENT') or is_granted('ROLE_BOARD_MEMBER') or is_granted('ROLE_COORDINATOR') or is_granted('ROLE_PROCURATION_MANAGER') or is_granted('ROLE_LEGISLATIVE_CANDIDATE') %}
+<div class="background--darker space--10-0 l__hide--on-small-laptop">
+    <ul class="l__wrapper list__links list__links--row list__links--row-left list__links--no-decor list__links--subnav nav-dropdown__menu__items">
+        {% if is_granted('ROLE_SUPERVISOR') or is_granted('ROLE_HOST')  %}
```
Comment (HeahDude) [inline templates/components/_nav_desktop.html.twig]: `app.user.isSupervisor` and `app.user.isHost`

File: templates/components/_nav_desktop.html.twig [hunk 33]
```diff
@@ -0,0 +1,213 @@
...
+            <li>
+                {{ al_committees }}
+            </li>
+        {% endif %}
+        {% if is_granted('ROLE_REFERENT') %}
```
Comment (HeahDude) [inline templates/components/_nav_desktop.html.twig]: `app.user.isReferent`

File: templates/components/_nav_desktop.html.twig [hunk 34]
```diff
@@ -0,0 +1,213 @@
...
+            <li>
+                <a href="{{ path('app_referent_users') }}">Espace référent</a>
+            </li>
+        {% endif %}
+        {% if is_granted('ROLE_BOARD_MEMBER') %}
```
Comment (HeahDude) [inline templates/components/_nav_desktop.html.twig]: `app.user.isBoardMember`

File: templates/components/_nav_desktop.html.twig [hunk 35]
```diff
@@ -0,0 +1,213 @@
...
+            <li>
+                <a href="{{ path('app_board_member_home') }}">Espace membre du Conseil</a>
+            </li>
+        {% endif %}
+        {% if is_granted('ROLE_COORDINATOR') %}
```
Comment (HeahDude) [inline templates/components/_nav_desktop.html.twig]: `app.user.isCoordinator`

File: templates/components/_nav_desktop.html.twig [hunk 36]
```diff
@@ -0,0 +1,213 @@
...
+            <li>
+                <a href="{{ path('app_coordinator_committees', { 's': constant('AppBundle\\Entity\\Committee::PENDING') }) }}">Espace coordinateur régional</a>
+            </li>
+        {% endif %}
+        {% if is_granted('ROLE_PROCURATION_MANAGER') %}
```
Comment (HeahDude) [inline templates/components/_nav_desktop.html.twig]: `app.user.isProductionManager`

File: templates/components/_nav_mobile.html.twig [hunk 37]
```diff
@@ -0,0 +1,212 @@
+<nav class="gtm__navigation__mobile l__wrapper l__hide--above-small-laptop space--20-0">
+    <ul class="list__links list__links--row list__links--nav l__row--h-stretch">
+        <li><h1><a href="{{ path('homepage') }}" class="link--no-decor">En Marche !</a></h1></li>
+    </ul>
+    <div class="nav-mobile">
+        <input type="checkbox" id="nav-menu">
+        <label for="nav-menu">Menu de navigation</label>
+        <div class="nav-mobile--menu">
+            {% if (not app.user) %}
```
Comment (HeahDude) [inline templates/components/_nav_mobile.html.twig]: useless parentheses

File: templates/components/_nav_mobile.html.twig [hunk 38]
```diff
@@ -0,0 +1,212 @@
...
+                    <li>
+                        <a href="{{ path('app_user_login') }}">Connexion</a>
+                    </li>
+                    <li>
+                        <a href="{{ path('app_membership_register') }}">Inscription</a>
+                    </li>
+                </ul>
+                <hr />
+            {% endif %}
+            {% if has_role_user %}
```
Comment (HeahDude) [inline templates/components/_nav_mobile.html.twig]: `else`

File: templates/components/_nav_mobile.html.twig [hunk 39]
```diff
@@ -0,0 +1,212 @@
...
+                {% endif %}
+                <ul>
+                    {% if has_role_adherent and is_granted('CREATE_COMMITTEE') %}
+                        <ul>
+                            <li>
+                                <a href="{{ path('app_adherent_create_committee') }}">Créer un comité</a>
+                            </li>
+                        </ul>
+                    {% endif %}
+                    {% if is_granted('ROLE_REFERENT') %}
```
Comment (HeahDude) [inline templates/components/_nav_mobile.html.twig]: Same is here and below, use `app.user.*` when possible

File: templates/components/_nav_mobile.html.twig [hunk 40]
```diff
@@ -0,0 +1,212 @@
...
+                </li>
+                <li>
+                    <a href="{{ path('articles_list', {category: 'tout'}) }}">Actualités</a>
+                </li>
+                {% if app.user and not has_role_adherent %}
+                    <li>
+                        <a href="{{ path('app_membership_register') }}">Devenir adhérent(e)</a>
+                    </li>
+                {% endif %}
+                {% if app.user and (has_role_adherent or is_granted('ROLE_HOST')) %}
```
Comment (HeahDude) [inline templates/components/_nav_mobile.html.twig]: `app.user.host`

File: templates/membership/register.html.twig [hunk 41]
```diff
@@ -45,13 +44,14 @@
...
-                    n'est pas l'addition d'individus, mais le rassemblement de personnes engagées qui
-                    partagent un socle de valeurs.</p>
+                <p class="text--body">Vous avez dès maintenant la possibilité d'adhérer à En Marche.
+                Nous sommes un collectif citoyen qui part d'une conviction : les Français doivent être au cœur
+                de la vie politique et non son décor.<br/>
+                Pour ce faire, veuillez remplir ces informations :</p>
             </article>
             <article class="l__wrapper--slim b__nudge--bottom-huge l__col l__col--center">
                 <div class="register__form">
+                    {% set isCreateForm = true %}
```
Comment (HeahDude) [inline templates/membership/register.html.twig]: Is this var used? should be named `is_create_form`

File: templates/components/_nav_desktop.html.twig (hunk 29 above)
Comment (B-Galati) [reply templates/components/_nav_desktop.html.twig]: Not the same, if you are admin you don't have this role.

File: templates/components/_nav_desktop.html.twig (hunk 30 above)
Comment (B-Galati) [reply templates/components/_nav_desktop.html.twig]: fixed

File: templates/components/_nav_desktop.html.twig (hunk 31 above)
Comment (B-Galati) [reply templates/components/_nav_desktop.html.twig]: fixed

File: templates/components/_nav_desktop.html.twig (hunk 34 above)
Comment (B-Galati) [reply templates/components/_nav_desktop.html.twig]: fixed

File: templates/components/_nav_desktop.html.twig (hunk 33 above)
Comment (B-Galati) [reply templates/components/_nav_desktop.html.twig]: fixed

File: templates/components/_nav_desktop.html.twig (hunk 36 above)
Comment (B-Galati) [reply templates/components/_nav_desktop.html.twig]: fixed

File: templates/components/_nav_desktop.html.twig (hunk 35 above)
Comment (B-Galati) [reply templates/components/_nav_desktop.html.twig]: fixed

File: templates/components/_nav_mobile.html.twig (hunk 37 above)
Comment (B-Galati) [reply templates/components/_nav_mobile.html.twig]: fixed

File: templates/components/_nav_mobile.html.twig (hunk 38 above)
Comment (B-Galati) [reply templates/components/_nav_mobile.html.twig]: No it is not the same condition.

File: templates/membership/register.html.twig (hunk 41 above)
Comment (B-Galati) [reply templates/membership/register.html.twig]: fixed

File: templates/components/_nav_mobile.html.twig (hunk 39 above)
Comment (B-Galati) [reply templates/components/_nav_mobile.html.twig]: fixed

File: templates/components/_nav_mobile.html.twig (hunk 40 above)
Comment (B-Galati) [reply templates/components/_nav_mobile.html.twig]: fixed

**Pull request merged**